code/game/machinery/computer/robot.dm

```diff
@@ -242,15 +243,23 @@
 				if(R.scrambledcodes)
 					return
 				// whatever weirdness this is supposed to be, but that is how the href gets added, so here it is again
-				if(istype(R) && istype(usr, /mob/living/silicon) && usr.mind.special_role && R.emagged != 1)
+				if(istype(R, /mob/living/silicon/robot) && istype(usr, /mob/living/silicon/ai) && ismalf(usr) && R.emagged != 1)
 					var/choice = input("Are you certain you wish to hack [R.name]?") in list("Confirm", "Abort")
 					if(choice == "Confirm")
-						if(R && istype(R))
-//							message_admins("<span class='notice'>[key_name_admin(usr)] emagged [R.name] using robotic console!</span>")
-							log_game("[key_name(usr)] emagged [R.name] using robotic console!")
-							R.SetEmagged(2)
-							if(R.mind.special_role)
-								R.verbs += /mob/living/silicon/robot/proc/ResetSecurityCodes
+						if(R)
+							if (!hacking)
+								hacking = 1
+								to_chat(usr, "Beginning override of cyborg safeties. This will take some time, and you cannot hack other borgs during the process.")
+								sleep(600)
+//								message_admins("<span class='notice'>[key_name_admin(usr)] emagged [R.name] using robotic console!</span>")
+								log_game("[key_name(usr)] emagged [R.name] using robotic console!")
+								R.SetEmagged(2)
```

Comment:
I hate this. This is the only part of the code that changes emagged to a non-binary value.

```suggestion
								R.SetEmagged(TRUE)
```

or

```suggestion
								R.illegal_weapons = TRUE
								R.SetEmagged()
```

The first one will make them pass emagged checks and the later will only give them the emagged modules. Pick one of these.